helpers/middlewares.js

```diff
@@ -7,14 +7,26 @@ exports.exceptionMiddleware = async function exceptionMiddleware(err, req, res,
     const pageId = err.pageId || req.query.pageId;
     const errorMessage = err.message || 'Unknown error';
 
-    res.status(500);
+
     if (contextId) {
         res.header('scrapy-puppeteer-service-context-id', contextId);
+    }  // What if contextId is undefined? Maybe we should skip such situations...
+
+    if (err.contextId) {  // there was a context, but something went wrong
+        res.status(500);
+
+        res.send({
+            contextId,
+            pageId,
+            error: errorMessage
+        });
+    } else {  // No context. Possibly, our service was restarted
+        res.status(422);
+
+        res.send({
+            error: errorMessage
+        });
     }
```

Comment:
It works well, but I'm not sure we should treat every error outside action calls as 422. For example, if a context/page creation fails, it is a server issue and we might want to reply with 500+. Maybe we should reply with 422 only on "context/page not found" errors? We can create a custom error type for that.